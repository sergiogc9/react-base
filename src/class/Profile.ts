export interface ProfileObject {}

export class Profile {

	// public user: User;

	// constructor(profile: ProfileObject) {
	// 	this.user = new User(profile.user);
	// }
}
