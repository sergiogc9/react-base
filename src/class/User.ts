export class UserObject {
	public id: string;
	public name: string;

	constructor(values?: Partial<UserObject>) {
		if (values) Object.assign(this, values);
	}
}

export class User extends UserObject {

}
