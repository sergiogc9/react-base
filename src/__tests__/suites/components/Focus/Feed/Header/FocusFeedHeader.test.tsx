import React from "react";
import { shallow, mount } from "enzyme";

import { FocusObject } from "@src/class/Focus";
import FocusFeedHeader from "@src/components/Focus/Feed/Header/FocusFeedHeader";

const focus: FocusObject = {
	deleted_at: new Date(1549287550563),
	feeds: { online: [], socialmedia: [], print: [] },
	id: "focus-id-1",
	inserted_at: new Date(1549287550563),
	name: "focus-name-1",
	updated_at: new Date(1549287550563),
	url_logo: "http://fake-url.com",
	acl_users: [
		{
			user_id: 'fake-user-id-1',
			discover_role: 'viewer',
			last_name: "Name",
			first_name: "User"
		},
		{
			user_id: 'fake-user-id-2',
			discover_role: 'viewer',
			last_name: 'User 2',
			first_name: 'Name 2'
		}
	]
};

const focus2: FocusObject = {
	...focus,
	id: "focus-id-2",
	name: "focus-name-2",
};

const focusList: FocusObject[] = [focus, focus2];

describe("FocusFeedHeader", () => {
	let wrapper: any;
	let onSaveFeed = jest.fn();
	let onSaveFeedName = jest.fn();
	let onSetEditNameInputValue = jest.fn();
	let onToggleFeedEnabled = jest.fn();
	let onToggleDrawer = jest.fn();
	let onDuplicateFeed = jest.fn();

	const getComponent = () => <FocusFeedHeader
		focusList={focusList}
		focusId={focus.id}
		feedId="feed-id"
		feedName="feed-name"
		feedType="socialmedia"
		feedEnabled={true}
		editNameInputValue={null}
		loadingSaveFeedName={false}
		feedChanged={false}
		showDuplicateFeedDialog={false}
		loadingSaveFeed={false}
		onSaveFeed={onSaveFeed}
		onSaveFeedName={onSaveFeedName}
		onSetEditNameInputValue={onSetEditNameInputValue}
		onToggleDrawer={onToggleDrawer}
		onToggleFeedEnabled={onToggleFeedEnabled}
		onDuplicateFeed={onDuplicateFeed}
	/>;

	beforeEach(() => {
		onSaveFeed = jest.fn();
		onSaveFeedName = jest.fn();
		onSetEditNameInputValue = jest.fn();
		onToggleFeedEnabled = jest.fn();
		onToggleDrawer = jest.fn();
		onDuplicateFeed = jest.fn();
		wrapper = shallow(getComponent());
	});

	it("Component snapshot", () => {
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Component snapshot while fetching data", () => {
		wrapper.setProps({ feedType: null });
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Component snapshot with show feed duplicate dialog", () => {
		wrapper.setProps({ showDuplicateFeedDialog: true });
		expect(wrapper.debug()).toMatchSnapshot();
	});

	it("Component snapshot with save feed loading", () => {
		wrapper.setProps({ loadingSaveFeed: true });
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Toggle drawer reponsive click", () => {
		wrapper.find('#focusFeedToggleDrawerBtn').at(0).simulate('click');
		expect(onToggleDrawer).toHaveBeenCalledTimes(1);
	});

	it("Component snapshot with edit name loading", () => {
		wrapper.setProps({ loadingSaveFeedName: true });
		expect(wrapper.html()).toMatchSnapshot();
		expect(wrapper.find('CircularProgress')).toHaveLength(1);
	});

	it("Component with edit name enabled", () => {
		wrapper.setState({ editNameEnabled: true });
		wrapper.setProps({ editNameInputValue: "name" });
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Component toggle edit name enabled handler", () => {
		wrapper.find('.edit-name-handler').at(0).simulate('click');
		expect(wrapper.state().editNameEnabled).toBe(true);
	});

	it("Component edit name input change event", () => {
		wrapper.setState({ editNameEnabled: true });
		wrapper.find('#focusFeedHeaderEditName').at(0).props().onChange("new text");
		expect(onSetEditNameInputValue).toHaveBeenCalledTimes(1);
		expect(onSetEditNameInputValue).toHaveBeenCalledWith("new text");
	});

	it("Component save feed click handlers", () => {
		wrapper = mount(getComponent());
		wrapper.setProps({ feedChanged: true });
		expect(onSaveFeed).toHaveBeenCalledTimes(0);
		wrapper.find('#focusFeedHeaderSaveFeedBtn').at(0).simulate('click');
		expect(onSaveFeed).toHaveBeenCalledTimes(1);
		wrapper.find('#focusFeedHeaderShowDrowpdownMenuResponsiveBtn').at(0).simulate('click');
		wrapper.find('div#focusFeedHeaderDropdownMenuResponsiveSaveFeed').at(0).simulate('click');
		expect(onSaveFeed).toHaveBeenCalledTimes(2);
	});

	it("Component edit name input blur event", () => {
		wrapper.setState({ editNameEnabled: true });
		wrapper.find('#focusFeedHeaderEditName').at(0).simulate('blur');
		expect(onSaveFeedName).toHaveBeenCalledTimes(1);
		expect(wrapper.state().editNameEnabled).toBe(false);
	});

	it("Component on enter key event", () => {
		wrapper.setState({ editNameEnabled: true });
		wrapper.find('#focusFeedHeaderEditName').at(0).simulate('keydown', { keyCode: 13 });
		expect(onSaveFeedName).toHaveBeenCalledTimes(1);
		expect(wrapper.state().editNameEnabled).toBe(false);
	});

	it("Component on other key event", () => {
		wrapper.setState({ editNameEnabled: true });
		wrapper.find('#focusFeedHeaderEditName').at(0).simulate('keydown', { keyCode: 10 });
		expect(onSaveFeedName).toHaveBeenCalledTimes(0);
		expect(wrapper.state().editNameEnabled).toBe(true);
	});

	it("Component snapshot with feed disabled", () => {
		wrapper.setProps({ feedEnabled: false });
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Component on toggle feed enabled", () => {
		wrapper.setState({ editNameEnabled: true });
		wrapper.find('#focusFeedHeaderToggleFeedEnabled').at(0).props().onChange();
		expect(onToggleFeedEnabled).toHaveBeenCalledTimes(1);
	});

	it("Component on toggle feed enabled in responsive menu", () => {
		wrapper = mount(getComponent());
		wrapper.find('button#focusFeedHeaderShowDrowpdownMenuResponsiveBtn').at(0).simulate('click');
		wrapper.find('div#focusFeedHeaderDropdownMenuResponsiveToggleEnableFeed').at(0).simulate('click');
		expect(onToggleFeedEnabled).toHaveBeenCalledTimes(1);
	});

	it("Component on duplicate button click handler", () => {
		wrapper = mount(getComponent());
		wrapper.find('button#focusFeedHeaderShowDrowpdownMenuBtn').at(0).simulate('click');
		wrapper.find('div#focusFeedHeaderDropdownMenuDuplicateFeed').at(0).simulate('click');
		expect(onDuplicateFeed).toHaveBeenCalledTimes(1);
	});

	it("Component focus on name input", () => {
		wrapper = mount(getComponent());
		wrapper.setState({ editNameEnabled: true });
		wrapper.update();
		expect(wrapper.html()).toMatchSnapshot();
	});
});
