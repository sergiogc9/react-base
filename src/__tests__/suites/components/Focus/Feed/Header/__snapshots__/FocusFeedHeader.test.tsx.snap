// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FocusFeedHeader Component focus on name input 1`] = `"<div id=\\"focusFeedHeader\\"><div class=\\"focus-basic-info\\"><button id=\\"focusFeedToggleDrawerBtn\\" type=\\"button\\" class=\\"md-btn md-btn--icon md-pointer--hover md-inline-block\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">filter_list</i></button><div class=\\"focus-name-link\\">focus.value: <a href=\\"/focus/focus-id-1\\">focus-name-1</a></div><i class=\\"md-icon material-icons\\">chevron_right</i><span>focus.social_feed: </span><div class=\\"md-text-field-container md-inline-block md-text-field-container--input focus-feed-header-edit-name\\"><label for=\\"focusFeedHeaderEditName\\" class=\\"md-floating-label md-floating-label--floating md-text--secondary\\">name</label><input id=\\"focusFeedHeaderEditName\\" placeholder=\\"name\\" type=\\"text\\" class=\\"md-text-field md-text-field--floating-margin md-text\\" value=\\"feed-name\\"><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-center\\"></div></div><div class=\\"focus-options\\"><div class=\\"md-selection-control-container md-switch-container\\"><input id=\\"focusFeedHeaderToggleFeedEnabled\\" type=\\"checkbox\\" class=\\"md-selection-control-input\\" name=\\"focusFeedHeaderToggleFeedEnabled\\" aria-hidden=\\"true\\" value=\\"\\" checked=\\"\\"><label for=\\"focusFeedHeaderToggleFeedEnabled\\" class=\\"md-selection-control-label md-pointer--hover md-text\\"><span id=\\"focusFeedHeaderToggleFeedEnabled-label\\">feed.status.active</span><div class=\\"md-switch-track md-pointer--hover md-switch-track--on\\"><div aria-labelledby=\\"focusFeedHeaderToggleFeedEnabled-label\\" role=\\"checkbox\\" aria-checked=\\"true\\" class=\\"md-fake-btn md-pointer--hover md-fake-btn--no-outline md-switch-thumb md-switch-thumb--on\\" tabindex=\\"0\\" aria-pressed=\\"false\\"><div class=\\"md-ink-container md-ink-container--2x\\"></div></div></div></label></div><div class=\\"menu-options\\"><div class=\\"menu-options-save\\"><button id=\\"focusFeedHeaderSaveFeedBtn\\" type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block\\"><div class=\\"md-ink-container\\"></div><div class=\\"md-icon-separator\\"><i class=\\"md-icon material-icons md-text--inherit\\">save</i><span class=\\"md-icon-text\\">feed.form.save</span></div></button></div><div class=\\"menu-options-dropdown\\"><div id=\\"focusFeedHeaderDropdownMenu\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-inline-block md-menu-container\\"><button id=\\"focusFeedHeaderShowDrowpdownMenuBtn\\" type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block\\"><div class=\\"md-ink-container\\"></div><div class=\\"md-icon-separator\\"><i class=\\"md-icon material-icons md-text--inherit\\">arrow_drop_down</i><div class=\\"md-icon-separator md-icon-text\\"><span class=\\"md-icon-text\\"></span></div></div></button></div></div></div><div class=\\"menu-options-responsive\\"><div id=\\"focusFeedHeaderDropdownMenuResponsive\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-inline-block md-menu-container\\"><button id=\\"focusFeedHeaderShowDrowpdownMenuResponsiveBtn\\" type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block\\"><div class=\\"md-ink-container\\"></div><div class=\\"md-icon-separator\\"><i class=\\"md-icon material-icons md-text--inherit\\">more_vert</i><div class=\\"md-icon-separator md-icon-text\\"><span class=\\"md-icon-text\\"></span></div></div></button></div></div></div></div>"`;

exports[`FocusFeedHeader Component snapshot 1`] = `"<div id=\\"focusFeedHeader\\"><div class=\\"focus-basic-info\\"><button id=\\"focusFeedToggleDrawerBtn\\" type=\\"button\\" class=\\"md-btn md-btn--icon md-pointer--hover md-inline-block\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">filter_list</i></button><div class=\\"focus-name-link\\">focus.value: <a href=\\"/focus/focus-id-1\\">focus-name-1</a></div><i class=\\"md-icon material-icons\\">chevron_right</i><span>focus.social_feed: </span><div class=\\"edit-name-handler\\"><span>feed-name</span><i class=\\"md-icon material-icons\\">edit</i></div></div><div class=\\"focus-options\\"><div class=\\"md-selection-control-container md-switch-container\\"><input type=\\"checkbox\\" id=\\"focusFeedHeaderToggleFeedEnabled\\" checked=\\"\\" class=\\"md-selection-control-input\\" name=\\"focusFeedHeaderToggleFeedEnabled\\" aria-hidden=\\"true\\"/><label for=\\"focusFeedHeaderToggleFeedEnabled\\" class=\\"md-selection-control-label md-pointer--hover md-text\\"><span id=\\"focusFeedHeaderToggleFeedEnabled-label\\">feed.status.active</span><div class=\\"md-switch-track md-pointer--hover md-switch-track--on\\"><div aria-labelledby=\\"focusFeedHeaderToggleFeedEnabled-label\\" role=\\"checkbox\\" aria-checked=\\"true\\" class=\\"md-fake-btn md-pointer--hover md-fake-btn--no-outline md-switch-thumb md-switch-thumb--on\\" tabindex=\\"0\\" aria-pressed=\\"false\\"><div class=\\"md-ink-container md-ink-container--2x\\"></div></div></div></label></div><div class=\\"menu-options\\"><div class=\\"menu-options-save\\"><button id=\\"focusFeedHeaderSaveFeedBtn\\" type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block\\"><div class=\\"md-ink-container\\"></div><div class=\\"md-icon-separator\\"><i class=\\"md-icon material-icons md-text--inherit\\">save</i><span class=\\"md-icon-text\\">feed.form.save</span></div></button></div><div class=\\"menu-options-dropdown\\"><div id=\\"focusFeedHeaderDropdownMenu\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-inline-block md-menu-container\\"><button id=\\"focusFeedHeaderShowDrowpdownMenuBtn\\" type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block\\"><div class=\\"md-ink-container\\"></div><div class=\\"md-icon-separator\\"><i class=\\"md-icon material-icons md-text--inherit\\">arrow_drop_down</i><div class=\\"md-icon-separator md-icon-text\\"><span class=\\"md-icon-text\\"></span></div></div></button></div></div></div><div class=\\"menu-options-responsive\\"><div id=\\"focusFeedHeaderDropdownMenuResponsive\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-inline-block md-menu-container\\"><button id=\\"focusFeedHeaderShowDrowpdownMenuResponsiveBtn\\" type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block\\"><div class=\\"md-ink-container\\"></div><div class=\\"md-icon-separator\\"><i class=\\"md-icon material-icons md-text--inherit\\">more_vert</i><div class=\\"md-icon-separator md-icon-text\\"><span class=\\"md-icon-text\\"></span></div></div></button></div></div></div></div>"`;

exports[`FocusFeedHeader Component snapshot while fetching data 1`] = `"<div id=\\"focusFeedHeader\\"></div>"`;

exports[`FocusFeedHeader Component snapshot with edit name loading 1`] = `"<div id=\\"focusFeedHeader\\"><div class=\\"focus-basic-info\\"><button id=\\"focusFeedToggleDrawerBtn\\" type=\\"button\\" class=\\"md-btn md-btn--icon md-pointer--hover md-inline-block\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">filter_list</i></button><div class=\\"focus-name-link\\">focus.value: <a href=\\"/focus/focus-id-1\\">focus-name-1</a></div><i class=\\"md-icon material-icons\\">chevron_right</i><span>focus.social_feed: </span><svg id=\\"focusFeedHeaderSavingFeedLoading\\" role=\\"progressbar\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" class=\\"md-progress md-progress--circular md-block-centered md-progress--circular-indeterminate\\" width=\\"18\\" height=\\"18\\" viewBox=\\"0 0 66 66\\"><circle class=\\"md-circular-progress-path md-circular-progress-path--animated\\" stroke-width=\\"6\\" stroke-linecap=\\"round\\" cx=\\"33\\" cy=\\"33\\" r=\\"30\\"></circle></svg></div><div class=\\"focus-options\\"><div class=\\"md-selection-control-container md-switch-container\\"><input type=\\"checkbox\\" id=\\"focusFeedHeaderToggleFeedEnabled\\" checked=\\"\\" class=\\"md-selection-control-input\\" name=\\"focusFeedHeaderToggleFeedEnabled\\" aria-hidden=\\"true\\"/><label for=\\"focusFeedHeaderToggleFeedEnabled\\" class=\\"md-selection-control-label md-pointer--hover md-text\\"><span id=\\"focusFeedHeaderToggleFeedEnabled-label\\">feed.status.active</span><div class=\\"md-switch-track md-pointer--hover md-switch-track--on\\"><div aria-labelledby=\\"focusFeedHeaderToggleFeedEnabled-label\\" role=\\"checkbox\\" aria-checked=\\"true\\" class=\\"md-fake-btn md-pointer--hover md-fake-btn--no-outline md-switch-thumb md-switch-thumb--on\\" tabindex=\\"0\\" aria-pressed=\\"false\\"><div class=\\"md-ink-container md-ink-container--2x\\"></div></div></div></label></div><div class=\\"menu-options\\"><div class=\\"menu-options-save\\"><button id=\\"focusFeedHeaderSaveFeedBtn\\" type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block\\"><div class=\\"md-ink-container\\"></div><div class=\\"md-icon-separator\\"><i class=\\"md-icon material-icons md-text--inherit\\">save</i><span class=\\"md-icon-text\\">feed.form.save</span></div></button></div><div class=\\"menu-options-dropdown\\"><div id=\\"focusFeedHeaderDropdownMenu\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-inline-block md-menu-container\\"><button id=\\"focusFeedHeaderShowDrowpdownMenuBtn\\" type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block\\"><div class=\\"md-ink-container\\"></div><div class=\\"md-icon-separator\\"><i class=\\"md-icon material-icons md-text--inherit\\">arrow_drop_down</i><div class=\\"md-icon-separator md-icon-text\\"><span class=\\"md-icon-text\\"></span></div></div></button></div></div></div><div class=\\"menu-options-responsive\\"><div id=\\"focusFeedHeaderDropdownMenuResponsive\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-inline-block md-menu-container\\"><button id=\\"focusFeedHeaderShowDrowpdownMenuResponsiveBtn\\" type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block\\"><div class=\\"md-ink-container\\"></div><div class=\\"md-icon-separator\\"><i class=\\"md-icon material-icons md-text--inherit\\">more_vert</i><div class=\\"md-icon-separator md-icon-text\\"><span class=\\"md-icon-text\\"></span></div></div></button></div></div></div></div>"`;

exports[`FocusFeedHeader Component snapshot with feed disabled 1`] = `"<div id=\\"focusFeedHeader\\"><div class=\\"focus-basic-info\\"><button id=\\"focusFeedToggleDrawerBtn\\" type=\\"button\\" class=\\"md-btn md-btn--icon md-pointer--hover md-inline-block\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">filter_list</i></button><div class=\\"focus-name-link\\">focus.value: <a href=\\"/focus/focus-id-1\\">focus-name-1</a></div><i class=\\"md-icon material-icons\\">chevron_right</i><span>focus.social_feed: </span><div class=\\"edit-name-handler\\"><span>feed-name</span><i class=\\"md-icon material-icons\\">edit</i></div></div><div class=\\"focus-options\\"><div class=\\"md-selection-control-container md-switch-container\\"><input type=\\"checkbox\\" id=\\"focusFeedHeaderToggleFeedEnabled\\" class=\\"md-selection-control-input\\" name=\\"focusFeedHeaderToggleFeedEnabled\\" aria-hidden=\\"true\\"/><label for=\\"focusFeedHeaderToggleFeedEnabled\\" class=\\"md-selection-control-label md-pointer--hover md-text\\"><span id=\\"focusFeedHeaderToggleFeedEnabled-label\\">feed.status.inactive</span><div class=\\"md-switch-track md-pointer--hover md-switch-track--off\\"><div aria-labelledby=\\"focusFeedHeaderToggleFeedEnabled-label\\" role=\\"checkbox\\" aria-checked=\\"false\\" class=\\"md-fake-btn md-pointer--hover md-fake-btn--no-outline md-switch-thumb md-switch-thumb--off\\" tabindex=\\"0\\" aria-pressed=\\"false\\"><div class=\\"md-ink-container md-ink-container--2x\\"></div></div></div></label></div><div class=\\"menu-options\\"><div class=\\"menu-options-save\\"><button id=\\"focusFeedHeaderSaveFeedBtn\\" type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block\\"><div class=\\"md-ink-container\\"></div><div class=\\"md-icon-separator\\"><i class=\\"md-icon material-icons md-text--inherit\\">save</i><span class=\\"md-icon-text\\">feed.form.save</span></div></button></div><div class=\\"menu-options-dropdown\\"><div id=\\"focusFeedHeaderDropdownMenu\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-inline-block md-menu-container\\"><button id=\\"focusFeedHeaderShowDrowpdownMenuBtn\\" type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block\\"><div class=\\"md-ink-container\\"></div><div class=\\"md-icon-separator\\"><i class=\\"md-icon material-icons md-text--inherit\\">arrow_drop_down</i><div class=\\"md-icon-separator md-icon-text\\"><span class=\\"md-icon-text\\"></span></div></div></button></div></div></div><div class=\\"menu-options-responsive\\"><div id=\\"focusFeedHeaderDropdownMenuResponsive\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-inline-block md-menu-container\\"><button id=\\"focusFeedHeaderShowDrowpdownMenuResponsiveBtn\\" type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block\\"><div class=\\"md-ink-container\\"></div><div class=\\"md-icon-separator\\"><i class=\\"md-icon material-icons md-text--inherit\\">more_vert</i><div class=\\"md-icon-separator md-icon-text\\"><span class=\\"md-icon-text\\"></span></div></div></button></div></div></div></div>"`;

exports[`FocusFeedHeader Component snapshot with save feed loading 1`] = `"<div id=\\"focusFeedHeader\\"></div>"`;

exports[`FocusFeedHeader Component snapshot with show feed duplicate dialog 1`] = `
"<div id=\\"focusFeedHeader\\">
  <div className=\\"focus-basic-info\\">
    <withInk(withTooltip(Button)) id=\\"focusFeedToggleDrawerBtn\\" icon={true} onClick={[Function: onClick]} inkTransitionOverlap={150} inkTransitionEnterTimeout={450} inkTransitionLeaveTimeout={300}>
      filter_list
    </withInk(withTooltip(Button))>
    <div className=\\"focus-name-link\\">
      focus.value
      : 
      <a href=\\"/focus/focus-id-1\\">
        focus-name-1
      </a>
    </div>
    <FontIcon iconClassName=\\"material-icons\\">
      chevron_right
    </FontIcon>
    <span>
      focus.social_feed
      : 
    </span>
    <div className=\\"edit-name-handler\\" onClick={[Function: onClick]}>
      <span>
        feed-name
      </span>
      <FontIcon iconClassName=\\"material-icons\\">
        edit
      </FontIcon>
    </div>
  </div>
  <div className=\\"focus-options\\">
    <SelectionControl id=\\"focusFeedHeaderToggleFeedEnabled\\" type=\\"switch\\" label=\\"feed.status.active\\" name=\\"focusFeedHeaderToggleFeedEnabled\\" checked={true} labelBefore={true} onChange={[Function: onChange]} checkedCheckboxIcon={{...}} uncheckedCheckboxIcon={{...}} checkedRadioIcon={{...}} uncheckedRadioIcon={{...}} />
    <div className=\\"menu-options\\">
      <div className=\\"menu-options-save\\">
        <withInk(withTooltip(Button)) id=\\"focusFeedHeaderSaveFeedBtn\\" flat={true} iconEl={{...}} onClick={[Function: onClick]} inkTransitionOverlap={150} inkTransitionEnterTimeout={450} inkTransitionLeaveTimeout={300}>
          feed.form.save
        </withInk(withTooltip(Button))>
      </div>
      <div className=\\"menu-options-dropdown\\">
        <DropdownMenu id=\\"focusFeedHeaderDropdownMenu\\" position=\\"br\\" menuItems={{...}} defaultVisible={false} repositionOnScroll={true} repositionOnResize={false}>
          <withInk(withTooltip(Button)) id=\\"focusFeedHeaderShowDrowpdownMenuBtn\\" flat={true} iconEl={{...}} inkTransitionOverlap={150} inkTransitionEnterTimeout={450} inkTransitionLeaveTimeout={300}>
            <IconSeparator label=\\"\\" component=\\"div\\" />
          </withInk(withTooltip(Button))>
        </DropdownMenu>
      </div>
    </div>
    <div className=\\"menu-options-responsive\\">
      <DropdownMenu id=\\"focusFeedHeaderDropdownMenuResponsive\\" position=\\"br\\" menuItems={{...}} defaultVisible={false} repositionOnScroll={true} repositionOnResize={false}>
        <withInk(withTooltip(Button)) id=\\"focusFeedHeaderShowDrowpdownMenuResponsiveBtn\\" flat={true} iconEl={{...}} inkTransitionOverlap={150} inkTransitionEnterTimeout={450} inkTransitionLeaveTimeout={300}>
          <IconSeparator label=\\"\\" component=\\"div\\" />
        </withInk(withTooltip(Button))>
      </DropdownMenu>
    </div>
  </div>
  <Connect(FocusFeedDuplicateFeedDialog) />
</div>"
`;

exports[`FocusFeedHeader Component with edit name enabled 1`] = `"<div id=\\"focusFeedHeader\\"><div class=\\"focus-basic-info\\"><button id=\\"focusFeedToggleDrawerBtn\\" type=\\"button\\" class=\\"md-btn md-btn--icon md-pointer--hover md-inline-block\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">filter_list</i></button><div class=\\"focus-name-link\\">focus.value: <a href=\\"/focus/focus-id-1\\">focus-name-1</a></div><i class=\\"md-icon material-icons\\">chevron_right</i><span>focus.social_feed: </span><div class=\\"md-text-field-container md-inline-block md-text-field-container--input focus-feed-header-edit-name\\"><label for=\\"focusFeedHeaderEditName\\" class=\\"md-floating-label md-floating-label--floating md-text--secondary\\">name</label><input type=\\"text\\" value=\\"name\\" id=\\"focusFeedHeaderEditName\\" placeholder=\\"name\\" class=\\"md-text-field md-text-field--floating-margin md-text\\"/><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-center\\"/></div></div><div class=\\"focus-options\\"><div class=\\"md-selection-control-container md-switch-container\\"><input type=\\"checkbox\\" id=\\"focusFeedHeaderToggleFeedEnabled\\" checked=\\"\\" class=\\"md-selection-control-input\\" name=\\"focusFeedHeaderToggleFeedEnabled\\" aria-hidden=\\"true\\"/><label for=\\"focusFeedHeaderToggleFeedEnabled\\" class=\\"md-selection-control-label md-pointer--hover md-text\\"><span id=\\"focusFeedHeaderToggleFeedEnabled-label\\">feed.status.active</span><div class=\\"md-switch-track md-pointer--hover md-switch-track--on\\"><div aria-labelledby=\\"focusFeedHeaderToggleFeedEnabled-label\\" role=\\"checkbox\\" aria-checked=\\"true\\" class=\\"md-fake-btn md-pointer--hover md-fake-btn--no-outline md-switch-thumb md-switch-thumb--on\\" tabindex=\\"0\\" aria-pressed=\\"false\\"><div class=\\"md-ink-container md-ink-container--2x\\"></div></div></div></label></div><div class=\\"menu-options\\"><div class=\\"menu-options-save\\"><button id=\\"focusFeedHeaderSaveFeedBtn\\" type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block\\"><div class=\\"md-ink-container\\"></div><div class=\\"md-icon-separator\\"><i class=\\"md-icon material-icons md-text--inherit\\">save</i><span class=\\"md-icon-text\\">feed.form.save</span></div></button></div><div class=\\"menu-options-dropdown\\"><div id=\\"focusFeedHeaderDropdownMenu\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-inline-block md-menu-container\\"><button id=\\"focusFeedHeaderShowDrowpdownMenuBtn\\" type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block\\"><div class=\\"md-ink-container\\"></div><div class=\\"md-icon-separator\\"><i class=\\"md-icon material-icons md-text--inherit\\">arrow_drop_down</i><div class=\\"md-icon-separator md-icon-text\\"><span class=\\"md-icon-text\\"></span></div></div></button></div></div></div><div class=\\"menu-options-responsive\\"><div id=\\"focusFeedHeaderDropdownMenuResponsive\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-inline-block md-menu-container\\"><button id=\\"focusFeedHeaderShowDrowpdownMenuResponsiveBtn\\" type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block\\"><div class=\\"md-ink-container\\"></div><div class=\\"md-icon-separator\\"><i class=\\"md-icon material-icons md-text--inherit\\">more_vert</i><div class=\\"md-icon-separator md-icon-text\\"><span class=\\"md-icon-text\\"></span></div></div></button></div></div></div></div>"`;
